internal/command/deploy/deploy.go

```diff
@@ -182,12 +182,14 @@ func determineAppConfig(ctx context.Context) (cfg *app.Config, err error) {
 
 			return
 		}
-
 		cfg.SetEnvVariables(parsedEnv)
 	}
 
 	if regionCode := flag.GetString(ctx, flag.RegionName); regionCode != "" {
-		cfg.SetPrimaryRegion(regionCode)
+		cfg.PrimaryRegion = regionCode
+	}
+	if cfg.PrimaryRegion != "" {
+		cfg.SetEnvVariable("PRIMARY_REGION", cfg.PrimaryRegion)
```

Comment:
Do we want to explicitly set this in config, or always set `PRIMARY_REGION` without exposing to the user? I could argue both ways, but think hiding it will make things less confusing if the primary region needs to change.

Reply:
Not entirely sure but I vote for setting PRIMARY_REGION env var dynamically and do not duplicate its purpose in fly.toml.

Reply:
Yup, sounds good to me.